### PR TITLE
force completes upgrade failure if operator failure

### PR DESCRIPTION
**Pull request by neisw**
Adding explicit signal that '[sig-cluster-lifecycle] Cluster completes upgrade' failed when preceding 'Cluster version operator acknowledges upgrade' fails before it.

### CODE REVIEW

File: test/e2e/upgrade/upgrade.go [hunk 1]
```diff
@@ -404,6 +404,11 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 			return nil, false
 		},
 	); err != nil {
+		//before returning the err force a failure for completes upgrade which follows this test
+		disruption.RecordJUnit(f, "[sig-cluster-lifecycle] Cluster completes upgrade", func() (error, bool) {
```
Comment (dgoodwin) [inline test/e2e/upgrade/upgrade.go:408]: Lets do a const for this string so we know we're  always matching the test name below.

File: test/e2e/upgrade/upgrade.go [hunk 2]
```diff
@@ -404,6 +404,11 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
...
+			framework.Logf("Cluster version operator failed to acknowledge upgrade request")
+			return fmt.Errorf("Cluster did not complete %s: %s", "upgrade", "operator failed to acknowledge upgrade request"), false
```
Comment (dgoodwin) [inline test/e2e/upgrade/upgrade.go:410]: No need for the formatting here just a static string should do the trick.

**Review by dgoodwin — COMMENTED** [review]
This looks like what I was expecting based on the slack convo.

File: test/e2e/upgrade/upgrade.go (hunk 2 above)
Comment (neisw) [reply test/e2e/upgrade/upgrade.go:410]: Went back and forth as errors.New wasn't imported and calling fmt without formatting felt odd too.  Will go with fmt.Errorf with a constant string for now, but can also pull in errrors if preferred.

**Comment by deads2k** [comment]
need better upgrade signal

/override ci/prow/e2e-gcp-upgrade
/override ci/prow/e2e-aws-fips

**Pull request merged**